### ISSUE TITLE
PCC drop in Transfuser due to Top-K indices.

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#2147]
### Describe the bug 

- Transfuser having pcc=-0.059461209923028946 . [nov13_transfuser_xla.log](https://github.com/user-attachments/files/23530171/nov13_transfuser_xla.log)
- Drop originates from this [sigmoid](https://github.com/tenstorrent/tt-forge-models/blob/8e32d1607ded717fa391d797b9d9ad1a67835434/transfuser/pytorch/src/model.py#L163C54-L163C64) & [conv2d](https://github.com/tenstorrent/tt-forge-models/blob/8e32d1607ded717fa391d797b9d9ad1a67835434/transfuser/pytorch/src/model.py#L128C13-L128C23) operations. 
- Running below sanity with saved inputs(Inputs to the problematic op saved during cpu fwd pass)  yields a pcc=0.8179553151130676 [nov13_sig_conv_saved_ip.log](https://github.com/user-attachments/files/23530170/nov13_zz_sig_conv_saved_ip.log)
- Both operations are independent, taking separate inputs.
- When run individually with the saved inputs, both ops pass with valid pcc. 
  - [nov13_sig_saved_ip.log](https://github.com/user-attachments/files/23530610/nov13_zf_sig_saved_ip.log)
  - [nov13_conv_saved_ip.log](https://github.com/user-attachments/files/23530611/nov13_zz_conv_saved_ip.log)
- When run together with random inputs, below sanity passed. [nov13_sig_conv_randn.log](https://github.com/user-attachments/files/23530622/nov13_zf_sig_conv_randn.log)
- This indicates that the **issue arises only when both operations are executed together using saved inputs**.

### Sanity 

```
class SigmoidConv(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv2d(
            in_channels=64,
            out_channels=64,
            kernel_size=(3, 3),
            stride=(1, 1),
            padding=(1, 1),
        )

    def forward(self, x1, x2):
        y1 = torch.sigmoid(x2)
        y2 = self.conv(x1)
        return (y1, y2)
```

### Steps to reproduce the issue

```
git checkout kkannan/nov13_transfuser_debug_xla
git submodule update --init --recursive
git lfs pull --include 'input_1.pt input_2.pt'

# model run
pytest tests/runner/test_models.py -k "transfuser and single_device-full-inference" -svv 

# sanity runs

# sigmoid + conv2d  - rand inputs
pytest tests/torch/single_chip/ops/test_sigmoid_and_conv.py::test_sigmoid_and_conv_randn -svv

# sigmoid _ conv2d - saved inputs
pytest tests/torch/single_chip/ops/test_sigmoid_and_conv.py::test_sigmoid_and_conv_saved_inputs -svv

# sigmoid - saved input
pytest tests/torch/single_chip/ops/test_sigmoid_and_conv.py::test_sigmoid_saved_inputs -svv

# conv2d - saved input
pytest tests/torch/single_chip/ops/test_sigmoid_and_conv.py::test_conv2d_saved_inputs -svv

```

**Comment by kamalrajkannan78** [comment]
**21/11/2025**


- Tried splitting the current sanity check into two separate sanities and observed that the PCC drop is caused by the incorrect PCC calculation logic.
- Full context:https://github.com/tenstorrent/tt-xla/issues/2270 
- On the latest main, the Transfuser model is currently failing with the following error, which blocks further validation:

```
TT_FATAL: Invalid head size: 54. The head size must be a multiple of the tile width (32)
```

- Ticket - https://github.com/tenstorrent/tt-mlir/issues/5886
- Log - [nov21_transfuser_xla.log](https://github.com/user-attachments/files/23676763/nov21_transfuser_xla.log)
- Because of this failure, I’m unable to verify how the suggested PCC calculation fix impacts the overall PCC results.

**Comment by kamalrajkannan78** [comment]
Able to replicate the drop in tt-metal - https://github.com/tenstorrent/tt-metal/issues/33432
